### DISCUSSION TITLE
cdc_msc example fails with "Device Descriptor Request Failed" after successfully enumerating the USB device

### DISCUSSION THREAD

**Discussion by vinayb-imdt** [hathach/tinyusb#1458] [General]
Hello,

I'm currently trying to run the cdc_msc_freertos example on a custom board that has the same USB controller (Synopsys/dwc2) as the STM32F23E Discovery Board. So far I have got the dcd_init() succeeding, IRQ handler, USB clock setup correctly and I see that the USB Mass storage device comes up in the Windows 10 device manager for a brief period and then after some time, I see that device manager is showing "Device Descriptor Request Failed". When the device was enumerated as Mass storage device I didn't have the removable drive come up, only device manager identified the device as Mass storage device. If I check the "hardware IDs", the VID is reported correctly which is 0xCAFE. 

I have collected the Wireshark logs and logs attached. I see that "GET MAX LUN reponse" from the device is reported as malformed [packet.] Any pointers on how to debug this issue or what is causing the mass storage device example to fail?

Wireshark Logs - 

[wireshark_capture.pdf](https://github.com/hathach/tinyusb/files/8627340/wireshark_capture.pdf)

Thanks

**Comment by hathach** [comment]
probably bsp/setup issue, you should compare the behavior and code with one of supported boards and try to change one line of code at a time to test it out.

**Reply by vinayb-imdt** [reply]
Many thanks @hathach .

It was indeed a setup issue. some of the defines required to be added in tusb_config.h and some tweaking to the way Integrated PHY was being setup for the custom board. I now have the CDC interface of the composite device working. Need to make changes to the MSC end-points to make it to work. Having CDC work is enough to prove that tinyUSB has been ported. 

I am now looking at how to enumerate the STM32F23E Discovery Board as a WinUSB device. The documentation on TinyUSB describes this - "Vendor-specific class support with generic In & Out endpoints. Can be used with MS OS 2.0 compatible descriptor to load winUSB driver without INF file." Where can I find examples or information on which descriptors need to be defined and how to implement it with tinyUSB? I have an inclination that I might have to use the Vendor device Class as done in webusb example but please let me know if any pointers to a working winUSB example are available ?

Thanks,

**Reply by hathach** [reply]
yeah, it is easier to start with an working software on working hardware. Regarding the WinUSB, it is already demonstrated in webusb_serial, just follow that example descriptor and check out MS docs for details.

**Reply by vinayb-imdt** [reply]
hello @hathach, I have started on the WebUSB_Serial example but now I'm facing some minor issues and need some advice on customising for a winUSB device, please have a look at - [https://github.com/hathach/tinyusb/discussions/](1465)

Thanks,
